pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -118,7 +118,8 @@ func (r *Reader) ReadToEnd(ctx context.Context) {
 }
 
 func (r *Reader) readHeader(ctx context.Context) (doneReadingFile bool) {
-	s := scanner.New(r, r.maxLogSize, r.initialBufferSize, r.Offset, r.headerSplitFunc)
+	// For headers use the default buffer size 16KB since they will not be usually larger than that.
```

Comment:
It will start at the default buffer size either way, but if your assumption is incorrect then this just adds allocations. I'd rather we just behave the same in both cases by starting at the default and growing as necessary.

Reply:
Let me rephrase. I think it should start with the _configured_ (or default if not configured) size, then grow as necessary.

Reply:
Perfect. That is what is doing now.